data/transactions/logic/eval.go

```diff
@@ -386,6 +381,11 @@ func eval(program []byte, cx *evalContext) (pass bool, err error) {
 	cx.stack = make([]stackValue, 0, 10)
 	cx.program = program
 
+	err = cx.checkRekeyAllowed()
```

Comment:
Is this check still required given that you already do it below in `check`?

Reply:
I don't think so, but there are several sanity checks done by both `check` and `eval`, and I don't want to break that pattern here.